x-pack/plugins/uptime/server/lib/requests/get_journey_details.ts

```diff
@@ -35,12 +40,18 @@ export const getJourneyDetails: UMElasticsearchQueryFn<
       },
     },
     size: 1,
+    _source: ['synthetics', 'monitor', '@timestamp'],
   };
 
   const { body: thisJourney } = await uptimeEsClient.search({ body: baseParams });
 
-  if (thisJourney?.hits?.hits.length > 0) {
-    const thisJourneySource: any = thisJourney.hits.hits[0]._source;
+  if (thisJourney.hits.hits.length > 0) {
+    const { _id, _source } = thisJourney.hits.hits[0];
+    const decodedJourneySource = JourneyStepType.decode(Object.assign({ _id }, _source));
+    if (!isRight(decodedJourneySource)) {
+      throw Error(`Could not process journey step for check group ${checkGroup}, malformed data`);
```

Comment:
Curious how this is handled on the server side and what the end user experience is. Will an error handler pick this up and send back an error status on the request, and will the UI fall back to no dat?

Reply:
The caller should catch the error and handle it as needed. I just checked the route that calls this and saw it's not catching the error, so I'll add code and check the rest of the `throw` sections I've added, as I may have missed others. In this case I tend to return `500` and pass the thrown string in the message.

If an unhandled `throw` occurs, Kibana will catch and log it:
![image](https://user-images.githubusercontent.com/18429259/123296897-0c116d00-d4e5-11eb-98ba-3dc01312c601.png)

Reply:
Implemented here: https://github.com/elastic/kibana/pull/101678/commits/219e2b3dd97b6596ffebacf64a87df3c4a120dcd.